File: domain/src/main/kotlin/com/kyovo/domain/service/RoomService.kt
package com.kyovo.domain.service

import com.kyovo.domain.model.NewRoom
import com.kyovo.domain.model.Room
import com.kyovo.domain.model.RoomId
import com.kyovo.domain.port.primary.RoomUseCase
import com.kyovo.domain.port.secondary.RoomRepository

class RoomService(private val roomRepository: RoomRepository) : RoomUseCase
{
    override fun findAll(): List<Room>
    {
        return roomRepository.findAll()
    }

    override fun findById(id: RoomId): Room?
    {
        return roomRepository.findById(id)
    }

    override fun save(newRoom: NewRoom): Room
    {
        return roomRepository.save(newRoom.toRoom())
    }
}

File: frontend/src/domain/service/RoomService.ts
import type { Room } from '@/domain/model/Room'
import type { RoomUseCase } from '@/domain/port/primary/RoomUseCase'
import type { RoomRepository } from '@/domain/port/secondary/RoomRepository'

export class RoomService implements RoomUseCase {
  constructor(private readonly repository: RoomRepository) {}

  findAll(): Promise<Room[]> {
    return this.repository.findAll()
  }

  findById(id: string): Promise<Room> {
    return this.repository.findById(id)
  }
}
